lib/gui/components/modal/services/modal.js

```diff
@@ -56,18 +56,29 @@ module.exports = function($uibModal, $q) {
       close: modal.close,
       result: $q((resolve, reject) => {
         modal.result
-          .then(resolve)
+          .then((value) => {
+            AnalyticsService.logEvent('Modal closed', {
+              method: 'User accepted',
+              value: value
+            });
+            resolve(value);
+          })
           .catch((error) => {
-
             // Bootstrap doesn't 'resolve' these but cancels the dialog;
             // therefore call 'resolve' here applied to 'false'.
             if (error === 'escape key press' || error === 'backdrop click') {
+              AnalyticsService.logEvent('Modal closed', {
+                method: error
+              });
               resolve();
 
             // For some annoying reason, UI Bootstrap Modal rejects
             // the result reason if the user clicks on the backdrop
             // (e.g: the area surrounding the modal).
-            } else if (error !== 'backdrop click') {
+            } else {
+              AnalyticsService.logEvent('Modal closed', {
```

Comment:
I don't think this corresponds to a "Click on X", but to any JavaScript error occurred on the modal. I might be wrong but I think a click on X causes the modal to go straight to `.then()`, while resolving it with an undefined value.